packages/node-runtime-worker-thread/src/lock.ts

```diff
@@ -0,0 +1,35 @@
+export type UNLOCKED = 'UNLOCKED';
+
+export class Lock {
+  private static UNLOCK_TOKEN: UNLOCKED = 'UNLOCKED';
+
+  private promise: Promise<UNLOCKED> | null = null;
+
+  private resolve: ((...args: any[]) => any) | null = null;
+
+  lock(): Promise<UNLOCKED> {
+    this.promise = new Promise((resolve) => {
```

Comment:
Should this make sure that it’s unlocked before?

Reply:
Yeah, good point, otherwise it can get locked forever in a confusing way, will change

Reply:
Fixed in 6680e42